fitbenchmarking/fitting_benchmarking.py

```diff
@@ -76,46 +74,41 @@ def _benchmark(user_input, problem_group, num_runs):
     Loops through software and benchmarks each problem within the problem
     group.
 
-    :param user_input :: all the information specified by the user
-    :type user_input :: UserInput
-    :param problem_group :: list of paths to problem files in the group
+    :param user_input: all the information specified by the user
+    :type user_input: fitbenchmarking.utils.user_input.UserInput
+    :param problem_group: Paths to problem files in the group
                           e.g. ['NIST/low_difficulty/file1.dat',
                                 'NIST/low_difficulty/file2.dat',
                                 ...]
-    :type problem_group :: list
-    :param num_runs :: number of times controller.fit() is run to
+    :type problem_group: list of string
+    :param num_runs: number of times controller.fit() is run to
                      generate an average runtime
-    :type num_runs :: str
-
-
-    :return :: array of result objects, per problem per user_input
-    :rtype :: list of FittingResult
+    :type num_runs: int
 
+    :returns: Result objects, per problem per user_input
+    :rtype: list of fitbenchmarking.plotting.plot_helper.data
     """
+    if not isinstance(user_input, list):
```

Comment:
Thanks for the pointer to EAFP, I didn't know this was the accepted preference for python (although you could probably guess that from bits like this).
In this case, I think it would be more problematic as the fact that it is a list (and not generator) is also used in a call to `len` on line 107.

I'm also a bit reluctant to change this as I know it is being changed when I do the options, which will render user_input redundant.

Reply:
You could just determine the `len` within the try/except, as both would raise `TypeError`.  Although really I think the problem could probably be avoided by flattening that loop.  However if you are going to remove `user_input` later then definitely don't worry about it.